### DISCUSSION TITLE
How can I use NativeDatabase to define SQLite functions with migration testing?

### DISCUSSION THREAD

**Discussion by cn-cn-cn** [simolus3/drift#2941] [Q&A] [answered]
I am using a SQLite custom function in my database class:
```dart
LazyDatabase _openConnection() {
    return LazyDatabase(() async {
     // ...
      return NativeDatabase(..., setup: (raw) {
          raw.createFunction(...);
    });
  });
```

How can I ensure this function gets defined even when using `InitializedSchema.newConnection` in a migration test?
I've read the interfaces, searched the web and these Discussions, but I haven't found a solution.

**Comment by simolus3** [comment, marked as answer]
This is not possible with a released drift version unfortunately, but I've just added the necessary callback in 37f120d287057c2cd2ad4b79f02a7584957b9f2.

**Reply by cn-cn-cn** [reply]
Thanks a lot!
